src/xbrl_extract/cli.py

```diff
@@ -14,83 +18,87 @@ def parse_main():
     )
     parser.add_argument(
-        "-t",
-        "--threads",
+        "-w",
+        "--workers",
         default=None,
         type=int,
-        help="Specify number of threads in pool (defaults to `os.cpu_count()`)"
-    )
-    parser.add_argument(
-        "--verbose",
-        action="store_true",
-        default=False,
-        help="Print logging messages to stdout",
+        help="Specify number of workers in pool (will attempt to choose a reasonable default if not specified)",
     )
     parser.add_argument(
         "--loglevel",
         help="Set log level",
         default="INFO",
     )
+    parser.add_argument("--logfile", help="Path to logfile", default=None)
 
     return parser.parse_args()
 
 
 def get_instances(instance_path: Path):
     """Get list of instances from specified path."""
-    ALLOWABLE_SUFFIXES = [".xbrl", ".xml"]
+    allowable_suffixes = [".xbrl", ".xml"]
 
     if not instance_path.exists():
-        raise ValueError("Must provide valid path to XBRL instance or directory"
-                         "of XBRL instances.")
+        raise ValueError(
+            "Must provide valid path to XBRL instance or directory" "of XBRL instances."
+        )
 
     # Single instance
     if instance_path.is_file():
-        if instance_path.suffix not in ALLOWABLE_SUFFIXES:
-            raise ValueError("Must provide valid path to XBRL instance or "
-                             "directory of XBRL instances.")
+        if instance_path.suffix not in allowable_suffixes:
+            raise ValueError(
+                "Must provide valid path to XBRL instance or "
+                "directory of XBRL instances."
+            )
 
-        return [(str(instance_path), 0)]
+        return [(str(instance_path), instance_path.name.rstrip(instance_path.prefix))]
     # Directory of instances
     elif instance_path.is_dir():
-        instances = [str(instance) for instance in instance_path.iterdir()
-                     if instance.suffix in ALLOWABLE_SUFFIXES]
-        return [(instance, i) for i, instance in enumerate(sorted(instances))]
+        return [
+            (str(instance), instance.name.rstrip(instance.suffix))
+            for instance in sorted(instance_path.iterdir())
+            if instance.suffix in allowable_suffixes
+        ]
 
 
 def main():
     """CLI for extracting data fro XBRL filings."""
     args = parse_main()
 
-    engine = create_engine(f"sqlite:///{args.to_sql}") if args.to_sql else None
+    logger = logging.getLogger("xbrl_extract")
+    logger.setLevel(args.loglevel)
+    log_format = "%(asctime)s [%(levelname)8s] %(name)s:%(lineno)s %(message)s"
+    coloredlogs.install(fmt=log_format, level=args.loglevel, logger=logger)
+
+    if args.logfile:
+        file_logger = logging.FileHandler(args.logfile)
+        file_logger.setFormatter(logging.Formatter(log_format))
+        logger.addHandler(file_logger)
+
+    engine = create_engine(f"sqlite:///{args.sql_path}")
```

Comment:
Later on you write the dataframes to the database using `pd.DataFrame.to_sql()`. [SQLAlchemy recommends](https://docs.sqlalchemy.org/en/14/core/connections.html#sqlalchemy.engine.Engine) using the python context manager so the database connection is closed once the dataframe is written.

Reply:
Fixed